From 1e2a920633d7085fc4e48f1163f434525b777c5b Mon Sep 17 00:00:00 2001
From: John Salatas <jsalatas@gmail.com>
Date: Wed, 6 Mar 2019 14:25:28 -0800
Subject: Preserve screen colour doesn't work as expected

Summary:
using a custom .icc profile, running redshift from a command line works good but the applet seems to mess with it, the preserve option doesn't work, either in randr mode or vidmode the .icc is ignored even if it is running in daytime.

See also:
https://github.com/kotelnik/plasma-applet-redshift-control/issues/54

Reviewers: #Plasma, martinkostolny, broulik

Reviewed By: broulik

Subscribers: broulik, plasma-devel

Tags: #plasma

Differential Revision: https://phabricator.kde.org/D17249
---
 package/contents/ui/main.qml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package/contents/ui/main.qml b/package/contents/ui/main.qml
index 2ea8fd3..268cb33 100644
--- a/package/contents/ui/main.qml
+++ b/package/contents/ui/main.qml
@@ -96,7 +96,7 @@ Item {
 
     function stopRedshift() {
         print('disabling redshift')
-        redshiftDS.connectedSources.length = 0
+        redshiftDS.removeSource(redshiftCommand)
         redshiftDS.connectedSources.push(redshiftDS.redshiftStopSource)
         active = false
     }
-- 
cgit v1.1
